src/scanner.jl

```diff
@@ -137,25 +137,98 @@ function get_mark(stream::TokenStream)
     Mark(stream.index, stream.line, stream.column)
 end
 
+# -------------
+# forwardchars!
+# -------------
 
-# Advance the stream by k characters.
-function forwardchars!(stream::TokenStream, k::Integer=1)
-    for _ in 1:k
+# Advance the stream by a chacater and the index.
+function forwardchar_skip!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    nothing
+end
+
+# Advance the stream by a chacater and the index and a column.
+function forwardchar_nobreak!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    stream.column += 1
+    nothing
+end
+
+# Advance the stream by a chacater and the index and break a line.
+function forwardchar_breakline!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    stream.column = 0
+    stream.line += 1
+    nothing
+end
+
+# YAML 1.1 [22] b-line-feed ::= #xA /*LF*/
+# YAML 1.2 [24] b-line-feed ::= x0A
+const b_line_feed = '\n'
+
+# YAML 1.1 [23] b-carriage-return ::= #xD /*CR*/
+# YAML 1.2 [25] b-carriage-return ::= x0D
+const b_carriage_return = '\r'
+
+# YAML 1.1 [24] b-next-line ::= #x85 /*NEL*/
+const yaml_1_1_b_next_line = '\u85'
+
+# YAML 1.1 [25] b-line-separator ::= #x2028 /*LS*/
+const yaml_1_1_b_line_separator = '\u2028'
+
+# YAML 1.1 [26] b-paragraph-separator ::= #x2029 /*PS*/
+const yaml_1_1_b_paragraph_separator = '\u2029'
+
+# Advance the stream by `n` characters.
+# YAML 1.1 [28] b-specific ::= b-line-separator | b-paragraph-separator
+yaml_1_1_is_b_specific(c::Char) = c == yaml_1_1_b_line_separator || c == yaml_1_1_b_paragraph_separator
+# YAML 1.1 [29] b-generic ::= ( b-carriage-return b-line-feed) | b-carriage-return | b-line-feed | b-next-line
+# YAML 1.1 [33] b-ignored-any ::= b-generic | b-specific
+function yaml_1_1_forwardchars!(stream::TokenStream, n::Integer=1)
```

Comment:
Thinking again about the design here - do we want to use dispatch? That is, instead of having totally diffeering functions, could we have something like

```
abstract type YAMLVersion end

struct YAMLV1_1 <: YAMLVersion end

forwardchars!(::YAMLV1_1, stream::TokenStream, n::Integer=1) #...
```

etc...

Reply:
Then we can even make a version that doesn't take the first argument, that dispatches to whichever version we want to be the default